pkg/builder/in_memory_build_queue.go

```diff
@@ -311,14 +313,28 @@ func (bq *InMemoryBuildQueue) Execute(in *remoteexecution.ExecuteRequest, out re
 			QueuedTimestamp: bq.getCurrentTime(),
 		}
 
-		span := trace.FromContext(ctx)
-		if span != nil {
-			desiredState.TraceContext = propagation.Binary(span.SpanContext())
-		}
+		// This is close enough to
+		// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+		// to use their span format.
+		//
+		// Note that ctx here is local to producing the injection, we
+		// don't otherwise use it.
+		ctx, span := bq.tracer.Start(ctx, "bb_worker send",
+			trace.WithSpanKind(trace.SpanKindProducer),
+			trace.WithAttributes(
+				label.String("messaging.system", "bb_scheduler"),
+				label.String("messaging.destination", instanceName.String()),
+				label.String("messaging.message_id", in.ActionDigest.Hash),
+			),
+		)
+		desiredState.W3CTraceContext = map[string]string{}
+		util.PropagateInjectMap(ctx, desiredState.W3CTraceContext)
```

Comment:
This would be easier to understand if it could just be written as something like:
```go
desiredState.W3CTraceContext = util.ExtractW3CTraceContextFromContext(ctx)
```
or something like that.

Reply:
Most of the change is in the other PR, but done.